core/src/main/java/software/amazon/awssdk/interceptor/ExecutionAttributes.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.interceptor;
+
+import java.util.HashMap;
+import java.util.Map;
+import software.amazon.awssdk.annotation.NotThreadSafe;
+import software.amazon.awssdk.annotation.SdkPublicApi;
+
+/**
+ * A mutable collection of {@link ExecutionAttribute}s that can be modified by {@link ExecutionInterceptor}s in order to save and
+ * retrieve information specific to the current execution.
+ *
+ * This is useful for sharing data between {@link ExecutionInterceptor} method calls specific to a particular execution.
+ */
+@SdkPublicApi
+@NotThreadSafe
+public class ExecutionAttributes {
```

Comment:
Yes, but it's more work for the customer than it needs to be. The other helper class requires a Class<T> to allow type-safe population with a Map<Key, ?>. We don't need that, so asking the customer to pass a Class<T> we don't use isn't ideal.